### DISCUSSION TITLE
Query about solax X1 Fit Gen4 Modbus Power Control

### DISCUSSION THREAD

**Discussion by mwrben** [wills106/homeassistant-solax-modbus#1124] [Q&A]
I've recently installed a Waveshare RS485 to RJ45 Ethernet to a solax x1 Fit Gen 4 (SN F43...). Thanks to the detailed guidance on this repository it all worked first time so thanks to everyone involved in this project. 

My question relates to Modbus Power Control. In my list of controls it shows number.solax_remotecontrol_autorepeat_duration and number.solax_remotecontrol_duration but none of the other relevant controls. I've looked in the plugin_solax.py file and my inverter is classed as "AC | GEN4 | X1 # RetroFit X1 3kW / 3.7kW". 

For example looking at the Button types in the file, remotecontrol_trigger has HYBRID | GEN4 | GEN5 as allowed_types, but remotecontrol_duration and remotecontrol_autorepeat_duration have allowed_types of AC | HYBRID | GEN4 | GEN5 which presumably explains why they show up. 

Does my inverter definitely not allow modbus power control or is it something that might work but hasn't been tried yet? If it's the latter I could edit the local plugin_solax.py file, add AC to the allowed_types and then report back.

**Comment by woutersioen31** [comment]
I have the same or a similar problem. I have a solax X1 retrofit gen4 (X1-Fit-5.0E) with modbus connection over LAN (apparently this model has native modbus over LAN support /LAN/DRM port). The remote control_trigger entity does not show up in the list of controls, but other controls (number.solax_remotecontrol_autorepeat_duration and number.solax_remotecontrol_duration) show up.

**Comment by wills106** [comment]
The current configuration of remote power control takes Solar PV into account in the calculations. But they aren't present in the Fit's, as they are storage only.

I don't know if they could be made to work in some form of power control?

**Reply by mwrben** [reply]
I had a look at plugin_solax.py and tried running it by adding AC to the relevant allowed_types line. Looking at 
def value_function_remotecontrol_recompute, I realised that PV wouldn't be available so I tried Enabled Power Control. When I clicked the trigger button, I could see that all the remotecontrol variables were showing readings in the dashboard - apart from remotecontrol_reactive_power and so remote control didn't turn on.

I'm only interested in being able to force charge the battery during the night and also at times turning discharge off in the run up to a peak tariff period. If that would work using Power Control and Enabled No Discharge and without  needing to define remotecontrol_reactive_power in the calculations, then I could try again by removing reactive_power from the function.
